auth0/resource_auth0_client.go

```diff
@@ -220,9 +219,56 @@ func readClient(d *schema.ResourceData, m interface{}) error {
 	if err != nil {
 		return err
 	}
-	d.SetId(c.ClientID)
+
 	d.Set("client_id", c.ClientID)
 	d.Set("client_secret", c.ClientSecret)
+	d.Set("name", c.Name)
+	d.Set("description", c.Description)
+	d.Set("app_type", c.AppType)
+	d.Set("logo_uri", c.LogoURI)
+	d.Set("is_first_party", c.IsFirstParty)
+	d.Set("oidc_compliant", c.OIDCConformant)
+	d.Set("callbacks", c.Callbacks)
+	d.Set("allowed_logout_urls", c.AllowedLogoutURLs)
+	d.Set("allowed_origins", c.AllowedOrigins)
+	d.Set("web_origins", c.WebOrigins)
+	d.Set("sso", c.SSO)
+	d.Set("sso_disabled", c.SSODisabled)
+	d.Set("cross_origin_auth", c.CrossOriginAuth)
+	d.Set("cross_origin_loc", c.CrossOriginLocation)
+	d.Set("custom_login_page_on", c.CustomLoginPageOn)
+	d.Set("custom_login_page", c.CustomLoginPage)
+	d.Set("custom_login_page_preview", c.CustomLoginPagePreview)
+	d.Set("form_template", c.FormTemplate)
+	d.Set("token_endpoint_auth_method", c.TokenEndpointAuthMethod)
+
+	if c.JWTConfiguration != nil {
+		var jwt_config = make(map[string]string)
+		mapstructure.Decode(c.JWTConfiguration, &jwt_config)
+		d.Set("jwt_configuration", jwt_config)
+	}
+
+	if c.EncryptionKey != nil {
+		var enc_key schema.Set
+		err := mapstructure.Decode(c.EncryptionKey, &enc_key)
+		if err != nil {
+			return err
+		}
+		d.Set("encryption_key", enc_key)
```

Comment:
Honestly I kind of prefer the way you had it previously (as long as it worked correctly). Did you verify whether the `schema.Set` is populated correctly using this method?

Reply:
It didn't work correctly :)

The acceptance tests worked well and showed that the approach I was using had problems.

This appears to be working correctly

Reply:
Disregard that

Reply:
I was getting JWTConfiguration and EncryptionKey muddled in my head - I have reverted my change as I agree